google-ads/src/main/java/com/google/ads/googleads/lib/GoogleAdsClient.java

```diff
@@ -143,15 +175,64 @@ public static Builder newBuilder() {
      */
     public abstract Builder setCredentials(Credentials credentials);
 
-    private void setCredentials(Properties properties) {
-      UserCredentials credentials =
-          UserCredentials.newBuilder()
-              .setClientId(properties.getProperty(ConfigPropertyKey.CLIENT_ID.getPropertyKey()))
-              .setClientSecret(
-                  properties.getProperty(ConfigPropertyKey.CLIENT_SECRET.getPropertyKey()))
-              .setRefreshToken(
-                  properties.getProperty(ConfigPropertyKey.REFRESH_TOKEN.getPropertyKey()))
-              .build();
+    private void setCredentials(Properties properties) throws IOException {
```

Comment:
This method is looking a little long - what do you think about extracting a method object?

Reply:
I revamped this a bit to separate out the validation step from the creation step. Also made the check for both types of keys more robust so it checks if _any_ key exists for each OAuth flavor.